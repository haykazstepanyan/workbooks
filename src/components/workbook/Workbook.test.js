import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { WorkbookForm } from "./";

describe("WorkbookForm component", () => {
	const mockStore = configureStore();
	const store = mockStore({
		workbook: {},
	});
	const { getByTestId } = render(
		<Provider store={store}>
			<WorkbookForm />
		</Provider>
	);
	const newWorkbook = getByTestId("workbook-form");
	it("should match component snapshot", () => {
		expect(newWorkbook).toMatchSnapshot();
	});
});
