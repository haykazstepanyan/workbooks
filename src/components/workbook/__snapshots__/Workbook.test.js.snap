// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkbookForm components should match component snapshot 1`] = `
<form
  class="forms"
  data-testid="workbook-form"
>
  <div
    class="inputs"
    data-testid="wrapper-box"
  >
    <input
      class="inputs__field"
      data-testid="input"
      id="firstName"
      pattern="[a-zA-Z]+"
      type="text"
      value=""
    />
    <label
      class="inputs__labels"
      data-testid="label"
      for="firstName"
    >
      First Name
    </label>
  </div>
  <div
    class="inputs"
    data-testid="wrapper-box"
  >
    <input
      class="inputs__field"
      data-testid="input"
      id="lastName"
      pattern="[a-zA-Z]+"
      type="text"
      value=""
    />
    <label
      class="inputs__labels"
      data-testid="label"
      for="lastName"
    >
      Last Name
    </label>
  </div>
  <div
    class="inputs"
    data-testid="wrapper-box"
  >
    <input
      class="inputs__field"
      data-testid="input"
      id="passport"
      type="text"
      value=""
    />
    <label
      class="inputs__labels"
      data-testid="label"
      for="passport"
    >
      Passport
    </label>
  </div>
  <div
    class="inputs"
    data-testid="wrapper-box"
  >
    <input
      class="inputs__field"
      data-testid="input"
      id="email"
      type="email"
      value=""
    />
    <label
      class="inputs__labels"
      data-testid="label"
      for="email"
    >
      Email
    </label>
  </div>
  <div
    class="inputs"
    data-testid="wrapper-box"
  >
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container"
      >
        <input
          class="inputs__field"
          id="birthDate"
          type="text"
          value="01/07/2021"
        />
      </div>
    </div>
    <label
      class="inputs__labels"
      data-testid="label"
      for="birthDate"
    >
      Birth date
    </label>
  </div>
  <div
    class="buttons_rightAlign"
  >
    <button
      class="buttons"
      data-testid="button"
      disabled=""
      type="submit"
    >
      Add
    </button>
  </div>
</form>
`;
